### ISSUE TITLE
Species /Genus identification

### ISSUE THREAD

**Issue by kassammohamed** [jtamames/SqueezeMeta#281]
Hello,

I would like to better understand the taxonomy assignement step, I am testing Sqeezemeta with public sets When I plot the N= 15 bacteria in the spices level I am always getting unclassified species (bacteriodes, etc). What would be the reason, % identification used `('species',85,'genus',60,'family',55,'order',50,'class',46,'phylum',42,'superkingdom',40);` ? database (I used the default NCBI nr) ?

Attached  plots for example.
![Capture](https://user-images.githubusercontent.com/65945304/114505873-97f45680-9c63-11eb-9ec7-fbd2f09d13df.PNG)
![Capture](https://user-images.githubusercontent.com/65945304/114506058-dbe75b80-9c63-11eb-8ebd-5f77a0be5ca2.PNG)

Thanks,

Mohamed

**Comment by fpusan** [comment]
Hi!

TL;DR: Not being able to classify reads down to the species level is perfectly normal in metagenomics datasets coming from environmetal samples (16S rRNA datasets are a different story). Any tool which reports species-level classification for such data is most likely lying to you.

The cutoffs were established [here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4005636/)
See the PDF manual ("The LCA Algorithm") for more details.
See also the [wiki](https://github.com/jtamames/SqueezeMeta/wiki/A-discussion-on-taxonomic-annotation-and-the-meaning-of-disparity-values-of-contigs)
See also my answers in:
- https://github.com/jtamames/SqueezeMeta/issues/172#issuecomment-706692940
- https://github.com/jtamames/SqueezeMeta/issues/187#issuecomment-722243188

for more insight on how taxonomic and functional annotation is performed in SqueezeMeta, and the philosophy guiding our design choices.

**Comment by jtamames** [comment]
Hello

In your case, I would make the plot at the genus level, and would use the "no_partial_classifications=T" option to get rid of the unclassified_whatever annotations.

Best,
J

**Comment by kassammohamed** [comment]
Hi Javier,

I have tried the option 'no_partial_classifications=T' but i am getting an error
`> plotTaxonomy(HA, rank='genus', count='percent', no_partial_classifications=T)
Error in colSums(data0[unclassified, ]) :
  'x' must be an array of at least two dimensions`

**Comment by fpusan** [comment]
Hi!
What is your version of SQMtools?
I can not reproduce the bug with the test data using the latest version.
Best,
Fernando

**Comment by kassammohamed** [comment]
Hi Fernando.

I amusing the version 

`> library('SQMtools')

This is SQMtools version 0.6.1

SQMtools is an open source software package distributed under GNU General
Public License version 3 (GPLv3). Details of GPLv3 is available at
http://www.gnu.org/licenses/gpl-3.0.html.

If using it in publications or products, please cite:

Puente-Sánchez F., García-García N. and Tamames J., SQMtools: automated
processing and visual analysis of ’omics data with R and anvi’o.
BMC Bioinformatics 21, 358 (2020). doi: 10.1186/s12859-020-03703-2.
`

**Comment by fpusan** [comment]
Weird, it should work there.
Any chance you can share the project with me?

**Comment by kassammohamed** [comment]
HI Fernando,

Sure how I can share with you by google drive is it ok ?

**Comment by fpusan** [comment]
Yes, that should do

**Comment by kassammohamed** [comment]
Hi Fernando 

do you need the entire folder of the project or just the results folder ?

**Comment by fpusan** [comment]
I would need the SqueezeMeta_conf.pl file, the results folder and the intermediate folder

**Comment by kassammohamed** [comment]
OK can you share with me the email to give the permission ?

**Comment by fpusan** [comment]
sure, send it to fpusan@gmail.com

**Comment by kassammohamed** [comment]
Hi Fernando
can you try 
https://drive.google.com/drive/folders/1My9zAfPjsx5scBemOr8c2h5MHQyScNgM?usp=sharing

**Comment by fpusan** [comment]
Hi!
It seems that the intermediate dir is missing some files. it only goes up to `10.*` but there should be a `19.*.` file in there

**Comment by kassammohamed** [comment]
Hi Fernando,

I am having the option --nocog --nokegg --nopfam --nobins --nomaxbin --nometabat

**Comment by fpusan** [comment]
Ah, ok.

**Comment by kassammohamed** [comment]
I only analyzed to get the Taxonomy information for the moment.
So to load the SQM project I had to used SQMlite in R package.

**Comment by fpusan** [comment]
Ah, ok. That's helpful to know, maybe the issue is related to `SQMlite` only.

**Comment by kassammohamed** [comment]
Indeed, I tried to load via SQM but it failed, maybe it needs all the files, as was missing the functional annotation.

**Comment by fpusan** [comment]
There are some fixes to that issue already implemented in our dev branch. I'll check whether your data loads there.

**Comment by kassammohamed** [comment]
OK sure  i will wait for your update thx

**Comment by fpusan** [comment]
Yeah, the problem was related to your dataset having only one sample. It is fixed now in the develop branch. As a bonus, the develop branch also can load your data using `loadSQM`.
So the easiest way to get it now would be:
1) Clone our repo `git clone http://github.com/jtamames/SqueezeMeta`
2) Inside the repo, move to the develop branch `git checkout develop`
3) **With the conda environment activated**, install the dev version of `SQMtools` with `R CMD INSTALL /path/to/SqueezeMeta-repo/lib/SQMtools`.

**Comment by kassammohamed** [comment]
Hi Fernanda,

thanks a lot it works.
I 've got 75% unclassified which is very huge at the species level.

**Comment by fpusan** [comment]
It is not very huge, unless you are working with an already-sequenced microorganism. Please see my first answer in this thread.
Fernando

**Comment by kassammohamed** [comment]
Hi 
I would like to reduce the sensitivity of unclassified species, what parameter i should change in the paramters.pl file

I would like to compare the results with other program that we have run previously



.

**Comment by fpusan** [comment]
I think @jtamames can help you with that.
May I ask what kind of samples are you working with?

**Comment by kassammohamed** [comment]
Hi Fernando,

I am working with human gut metagenomics samples

**Comment by fpusan** [comment]
Now that I think of it, maybe I can help too.
Try loading your data into SQMtools with
`HA=loadSQM("/path/to/your/project", tax_mode="nofilter")`

**Comment by kassammohamed** [comment]
Hi Fernando 

I am getting this error

> HA=loadSQM("Sample1", tax_mode='nofilter')
Error in loadSQM("Sample1", tax_mode = "nofilter") :
  tax_mode must be either "allfilter" (apply minimum identity threshold for all taxa) or "prokfilter" (don't apply thresholds to Eukaryotes)

**Comment by fpusan** [comment]
Oh, I guess we don't allow for it since it's actually not a very good idea...
The table actually exists, but you'll have to look for it manually.
It is in `/path/to/your/project/results/tables/*.species.nofilter.abund.tsv`

**Comment by jtamames** [comment]
Hello
What do you mean by "reduce the sensitivity of unclassified species"? Having more annotations? I would advise to focus on quality, not in quantity. But you can change the parameters affecting the Diamond search (all identified with STEP4 in the parameters.pl file), the ones controlling the LCA algorithm (STEP6/8), and the ones for contig annotation (STEP9). The explanation of each one of these would be lenghty and it is already done in the manual, so please refer to it for details.

Best,

**Comment by kassammohamed** [comment]
Dear Javier and Fernanda,

Thank your or your inputs.
I was trying to link the method from squeezemeta to sqm2table and I would like to better understand what file is used to make the table because I see some difference and the creation of unclassified genus and species ?

**Comment by fpusan** [comment]
Hi!
It is a bit complex, since the script generates a lot of tables from a lot of files.
What table in particular are you interested in?
